src/errors.ts

```diff
@@ -0,0 +1,82 @@
+import { register } from "./core";
+import { instrument } from "./instrumentation";
+
+let regexEscape = (str) => str.replace(/([.?*+^$&[\]\\(){}|<>-])/g, "\\$1");
+
+export function logError(errorToLog) {
```

Comment:
logError is exposed for two reasons - the first is that it enables consumers of clarity to call logerror from their code if they want to log information without throwing exceptions + interfering with execution of the page. 

The second is that it enables testability through phantomjs - phantomjs doesn't support firing custom events but I wanted to trigger the execution in a way the page would actually interact with the method.